ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -315,6 +333,21 @@ private void initiateChunkDeletesIfAny(final String blobId, final String service
     backgroundDeleter.getBlob(blobId, options, callback);
   }
 
+  /**
+   * Completes a getBlob operation by invoking the {@code callback} and setting the {@code futureResult} with the given
+   * {@code {@link RouterException}}
+   * @param routerException {@link RouterException} to be set in the callback and future result
+   * @param internalOptions instance of {@link GetBlobOptionsInternal} to use
+   * @param futureResult the {@link FutureResult} that needs to be set.
+   * @param callback that {@link Callback} that needs to be invoked. Can be null.
+   */
+  private void completeGetBlobOperation(RouterException routerException, GetBlobOptionsInternal internalOptions,
+      FutureResult<GetBlobResult> futureResult, Callback<GetBlobResult> callback) {
+    routerMetrics.operationDequeuingRate.mark();
+    routerMetrics.onGetBlobError(routerException, internalOptions, false);
```

Comment:
it seems like sometimes you _do_ have access to whether the blob is encrypted or not when you call this function, no?